app/service/contact_svc.py

```diff
@@ -75,6 +76,11 @@ async def _save(self, result):
                 link.status = int(result.status)
                 if result.output:
                     link.output = True
+                    # check if the ability has any applicable post processing hooks
+                    if link.ability.HOOKS and link.ability.executor in link.ability.HOOKS:
+                        processed_output = await self._postprocess_link_result(result.output, link.ability)
+                        if processed_output:
+                            result.output = processed_output
```

Comment:
I'd move this all into the post process link result function to keep _save clean